### PR TITLE
More performance enhancements

### PR DESCRIPTION
**Pull request by seeker25**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

File: pay-api/src/pay_api/models/payment.py [hunk 1]
```diff
@@ -379,17 +379,20 @@ def get_invoices_and_payment_accounts_for_statements(cls, search_filter: Dict):
     @classmethod
     def get_count(cls, auth_account_id: str, search_filter: Dict):
         """Slimmed downed version for count (less joins)."""
-        query = cls.generate_base_transaction_query()
-        query = cls.filter(query, auth_account_id, search_filter)
+        query = db.session.query(Invoice.id)
```
Comment (seeker25) [inline pay-api/src/pay_api/models/payment.py:382]: Too many joins previously, when they end up getting eliminated through group_by(Invoice.id)

File: pay-api/src/pay_api/models/payment.py [hunk 2]
```diff
@@ -505,9 +514,9 @@ def filter_details(cls, query, search_filter: dict):
     @classmethod
     def generate_subquery(cls, auth_account_id, search_filter, limit, page):
         """Generate subquery for invoices, used for pagination."""
-        subquery = cls.generate_base_transaction_query()
+        subquery = db.session.query(Invoice.id)
```
Comment (seeker25) [inline pay-api/src/pay_api/models/payment.py:517]: Too many joins previously, when they end up getting eliminated through group_by(Invoice.id)

File: pay-api/src/pay_api/models/payment.py [hunk 3]
```diff
@@ -289,7 +289,7 @@ def generate_base_transaction_query(cls):
                 InvoiceReference.reference_number,
                 InvoiceReference.status_code,
             )
-            .outerjoin(PaymentAccount, Invoice.payment_account_id == PaymentAccount.id)
+            .join(PaymentAccount, Invoice.payment_account_id == PaymentAccount.id)
```
Comment (seeker25) [inline pay-api/src/pay_api/models/payment.py:292]: This large query is our main query, tried to narrow down our count and subqueries so they come up a bit more simple

**Comment by seeker25** [comment]
@Jxio Looks like you're right about the counts:

With count:
image

Without count:
image

File: pay-api/src/pay_api/models/payment.py [hunk 4]
```diff
@@ -252,50 +251,62 @@ def find_payments_to_consolidate(cls, auth_account_id: str):
...
-                InvoiceReference.id,
-                InvoiceReference.invoice_number,
-                InvoiceReference.reference_number,
-                InvoiceReference.status_code,
-            )
-            .outerjoin(PaymentAccount, Invoice.payment_account_id == PaymentAccount.id)
-            .outerjoin(PaymentLineItem, PaymentLineItem.invoice_id == Invoice.id)
-            .outerjoin(
+            db.session.query(Invoice)
+            .join(PaymentAccount, Invoice.payment_account_id == PaymentAccount.id)
```
Comment (seeker25) [inline pay-api/src/pay_api/models/payment.py:255]: Joins make sense for these, they will always occur

**Comment by seeker25** [comment]
For Monday:

Needs quick side by side comparison
Pay Jobs CI fixed

**Review by Jxio — APPROVED** [review]

**Review by ochiu — APPROVED** [review]

**Pull request merged**